cmake/modules/extensions.cmake

```diff
@@ -493,6 +493,11 @@ function(zephyr_library_compile_definitions item)
   target_compile_definitions(${ZEPHYR_CURRENT_LIBRARY} PRIVATE ${item} ${ARGN})
 endfunction()
 
+# Similar as above functions support add_dependencies
+function(zephyr_library_add_dependencies)
```

Comment:
nit: Since this group of functions is described as:

> zephyr_library versions of normal CMake **target_\<func\>** functions

maybe this should be placed underneath `zephyr_library_cc_option`.